code/implementation/system-stack/src/main/java/io/cattle/platform/systemstack/listener/external/api/catalog/Template.java

```diff
@@ -0,0 +1,376 @@
+package io.cattle.platform.systemstack.listener.external.api.catalog;
+
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class Template {
+    @JsonProperty("catalogId")
+    private String catalogId;
```

Comment:
There's no particular reason to make all the variables as private.  I would just leave them as the default visibility.